### PR TITLE
[release-1.2] fix: do not clear out spec.Domain.Memory

### PR DESCRIPTION
**Pull request by acardace**
Do not clear out `Memory` in `setGuestMemory` but rather overwrite the `Guest` field, otherwise if hugepages are configured for the VM the setting would get lost.

### What this PR does
Before this PR:
Creating a VM configured with hugepages does not actually use hugepages if memory is set through `memory.requests`

After this PR:
Creating a VM configured with hugepages does use hugepages if memory is set through `memory.requests`

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-controller/watch/vm.go [hunk 1]
```diff
@@ -3457,15 +3457,15 @@ func setGuestMemory(spec *virtv1.VirtualMachineInstanceSpec) {
...
-		spec.Domain.Memory = &virtv1.Memory{
-			Guest: spec.Domain.Resources.Requests.Memory(),
-		}
+		spec.Domain.Memory.Guest = spec.Domain.Resources.Requests.Memory()
 	case !spec.Domain.Resources.Limits.Memory().IsZero():
-		spec.Domain.Memory = &virtv1.Memory{
-			Guest: spec.Domain.Resources.Limits.Memory(),
-		}
+		spec.Domain.Memory.Guest = spec.Domain.Resources.Limits.Memory()
 	case spec.Domain.Memory != nil && spec.Domain.Memory.Hugepages != nil:
```
Comment (fossedihelm) [inline pkg/virt-controller/watch/vm.go:3469]: I think we can drop `spec.Domain.Memory != nil` now

**Review by fossedihelm — COMMENTED** [review]
Can we add a unit test?

File: pkg/virt-controller/watch/vm.go (hunk 1 above)
Comment (acardace) [reply pkg/virt-controller/watch/vm.go:3469]: ah yes, you're right.

**Review by xpivarc — COMMENTED** [review]
/approve

**Pull request merged**